codeHF/PlotEfficiency_recoStep.C

```diff
@@ -0,0 +1,395 @@
+// Plotting of reconstruction efficiency
+
+#include "utils_plot.h"
+
+Int_t PlotEfficiency(TString pathFile = "AnalysisResults.root", TString particles = "d0", bool effSbyS = true)
+{
+  gStyle->SetOptStat(0);
+  gStyle->SetPalette(0);
+  gStyle->SetCanvasColor(0);
+  gStyle->SetFrameFillColor(0);
+
+  // vertical margins of the pT spectra plot
+  float marginHigh = 0.05;
+  float marginLow = 0.05;
+  bool logScaleH = true;
+  // vertical margins of the efficiency plot
+  float marginRHigh = 0.05;
+  float marginRLow = 0.05;
+  bool logScaleR = false;
+  double yMin, yMax;
+  int nRec, nGen;
+  int colours[] = {1, 2, 3, 4};
+  int markers[] = {24, 25, 46};
+
+  // binning
+  Int_t iNRebin = 4;
+  //Double_t* dRebin = nullptr;
+  //const Int_t NRebin = 1;
+  Double_t dRebin[] = {0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5, 4, 5, 6, 8, 10};
+  const Int_t NRebin = sizeof(dRebin) / sizeof(dRebin[0]) - 1;
+
+  TFile* file = new TFile(pathFile.Data());
+  if (file->IsZombie()) {
+    Printf("Error: Failed to open file %s", pathFile.Data());
+    return 1;
+  }
+
+  // get list of particles
+  TObjArray* arrayParticle = 0;
+  arrayParticle = particles.Tokenize(" ");
+  TString particle = "";
+
+  // compute efficiency at each reconstruction step
+  TString recostep[4] = {"RecoHFFlag", "RecoTopol", "RecoCand", "RecoPID"};
+  const int NRecoStep = sizeof(recostep) / sizeof(recostep[0]);
+
+  TCanvas* cgensbys = new TCanvas("cgensbys", "cgensbys", 800, 600);
+  if (NRecoStep == 4)
+    cgensbys->Divide(2, 2);
+
+  TCanvas* cgensbys_pt = new TCanvas("cgensbys_pt", "cgensbys_pt", 800, 600);
+  if (NRecoStep == 4)
+    cgensbys_pt->Divide(2, 2);
+
+  TCanvas* cgensbys_y = new TCanvas("cgensbys_y", "cgensbys_y", 800, 600);
+  if (NRecoStep == 4)
+    cgensbys_y->Divide(2, 2);
```

Comment:
I see a lot of repetition here. This can go in a single block.

Reply:
Maybe even in a loop.

Reply:
I have put the Divide in a single if-statement to simplify the code.
The loop option could be used moving to an array of TCanvas.